### DISCUSSION TITLE
Unable to link db file with SQL server

### DISCUSSION THREAD

**Discussion by rik1301** [sqlitebrowser/sqlitebrowser#3444] [Q&A]
Hi, my website dev is saying he doesn't know how to get the .db file to talk with SQL.  Instead he is trying to do some convulted way of exporting the .db file to csv file first, then reimporting it in SQL via phpmyadmin.   When I had my database query website running some years ago before I let it lapse, the original dev set it up so all I had to do was drag the .db file to my website FTP and 10 seconds later once it had finished uploading, the fresh query was ready to be queried on the front-end site.  So I know it's possible to do, but this dev doesn't seem to know what to do.  I'm not clued up on the technicalilites of it all, but a quick search online for .db files and SQL servers took me to stackexchange where there are several conversations about it not being possible to do without some ODBC/SQL connection/driver thingy.  

Can anyone expand on the issue and what to do to fix it?  Or better still would anyone be kind enough to offer to sort it for me in exchange for some beer tokens?   Everything is configured properlly and the website database query page is working perfectly, but currently requires me to export the database as csv, then log in to my cPanel, go to phpmyadmin, click on my database and then import the freshly updated csv file, which is a lot of messing around.  Instead all I want to do is just drag the .db file to my FTP.

Thanks for reading.

**Comment by FriedrichFroebel** [comment]
This seems to  be a rather specific issue and without digging too deep, I agree that a direct way probably is not possible without doing additional scripting.

DB4S (or *sqlitebrowser*), where you opened this discussion, works with SQLite3 databases. It seems like you have a `.db` file which itself contains a SQLite3 database. SQLite3 database file are standalone databases in a single file.

phpMyAdmin on the other hand allows you to administrate MySQL and MariaDB databases, a server-based database system. SQLite3 and MySQL generally are incompatible and one easy way to make them compatible is with the CSV-based export/import you mention.

You are not telling anything about the website technologies itself, but for me it sounds like in the past the website used an SQLite3 database, but has been migrated to MySQL later. Depending on the actual website code, it might be possible to replace the database backend to use SQLite3 (again) without much overhead, although SQLite3 databases usually are not recommended for production-grade websites (at least if the traffic is high enough).

Depending on your server itself, you might be able to listen on changes to an uploaded SQLite3 database file (on Linux, this might use the *inotify* kernel API) and start an automatic conversion to MySQL as well to avoid the manual conversion, but this rather seems like a hack. This might require you to have SSH access to the server instead of just FTP (SSH access is preferable for security reasons anyway, as FTP usually is using plaintext, but this is just a side note) to set everything up.

Having said this, you request seems to be a bit out of scope for DB4S itself and might fit better on some general programming Q&A platform like Stackoverflow and is rather vague as well.

**Reply by rik1301** [reply]
The database was created in DB4S from scratch.  It's just a single level database (basically an excel sheet with powerful column filters) and I do all the editing in the browser view.  I've done it this way for years and it was never an issue.  When I click the save project button each day, I get a .sqbpro file (which keeps all my column widths in place) and also a .db file which is the same, but with the default column widths.  I've never seen any other file extension.   I know it's quite possible to link a .db file to SQL though the FTP because that's exactly how my old site functioned - a simple drag and drop of the .db file to the FTP.  There was no converting or exporting to different file types.

**Reply by FriedrichFroebel** [reply]
As already mentioned: phpMyAdmin indicates that at some place in time, your website has been reworked to now use a MySQL database, which is incompatible with SQLite3 databases without further work. Thus there is not much we can do here on the DB4S side. You probably should discuss such issues with your developer/contractor and let him explain the current status in a way that you actually understand what they are talking about and how a suitable solution can be implemented such that both sides are happy about it.

**Reply by justinclift** [reply]
@rik1301 It sounds (to me) like you need to have a chat with your developer, and ask them to familiarise themselves with SQLite databases.  They're clearly familiar with MySQL already to some extent, which is why they're trying to use the MySQL based admin stuff.

Most (but not all :frowning:) website frameworks can understand SQLite by default, so (in theory) it shouldn't be too hard to have yours be adjusted for SQLite.  That way, the developer could completely drop the need for mysqladmin.

That being said, there might be a mismatch between your needs and the developer you have too (no idea). :frowning:
